gqcp/include/Processing/Properties/DysonOrbital.hpp

```diff
@@ -0,0 +1,155 @@
+
+            passive_mod1 = 1;
+            passive_mod2 = 1;
+            target_mod = passive_onv_basis1.dimension();
+        }
+
+        const auto& ci_coeffs1 = linear_expansion1.coefficients();
+        const auto& ci_coeffs2 = linear_expansion2.coefficients();
+
+        VectorX<double> dyson_coeffs = VectorX<double>::Zero(onv_basis1.alpha().numberOfOrbitals());
+
+        // The actual algorithm to determine the Dyson amplitudes
+        // Since we want to calculate the overlap between two wave functions, the ONVs should have an equal number of electrons
+        // We therefore iterate over the ONVs of the 'target' ONV basis, which all have an electron more, and annihilate in one of the orbitals (let the index of that orbital be called 'p')
+        // By calculating the overlap in the (N-1)-ONV basis, we can calculate the contributions to the  'p'-th coefficient (i.e. the Dyson amplitude) of the Dyson orbital
+        SpinUnresolvedONV onv = target_onv_basis1.constructONVFromAddress(0);
+
+        for (size_t It = 0; It < target_onv_basis1.dimension(); It++) {           // It loops over addresses of the target ONV basis
+            int sign = -1;                                                        // total phase factor of all the annihilations that have occurred
+            for (size_t e = 0; e < target_onv_basis1.numberOfElectrons(); e++) {  // loop over electrons in the ONV
+
+                // Annihilate on the corresponding orbital, to make sure we can calculate overlaps in the (N-1)-'target' ONV basis
+                sign *= -1;
+                size_t p = onv.occupationIndexOf(e);
+                onv.annihilate(p);
+
+                // Now, we calculate the overlap in the (N-1)-'target' ONV basis
+                // In order to access the correct coefficients for the, we need the address of the resulting (annihilated) ONV inside the 'target' ONV basis
+                size_t address = target_onv_basis2.addressOf(onv.unsignedRepresentation());
+
+                double coeff = 0;
+                for (size_t Ip = 0; Ip < passive_onv_basis1.dimension(); Ip++) {  // Ip loops over the addresses of the passive ONV basis
+
+                    coeff += sign * ci_coeffs1(It * target_mod + Ip * passive_mod1) * ci_coeffs2(address * target_mod + Ip * passive_mod2);  // access the indices of the coefficient vectors
+                }
+                dyson_coeffs(p) += coeff;
+                onv.create(p);  // allow the iteration to continue with the original ONV
+            }
+
+            if (It < target_onv_basis1.dimension() - 1) {  // prevent the last permutation from occurring
+                target_onv_basis1.transformONVToNextPermutation(onv);
+            }
+        }  // target address (It) loop
+
+        return DysonOrbital<Scalar>(dyson_coeffs);
+    }
+
+    /*
+     *  MARK: Access
+     */
+
+    /**
+     *  @return The expansion coefficients of this Dyson orbital.
```

Comment:
This requires some more documentation: what does each index of this vector represent?